provider/infoblox/infoblox.go

```diff
@@ -79,13 +89,15 @@ func NewMaxResultsRequestBuilder(maxResults int) *MaxResultsRequestBuilder {
 }
 
 // BuildRequest prepares the api request. it uses BuildRequest of
-// WapiRequestBuilder and then add the _max_requests parameter
+// WapiRequestBuilder and then add the _max_requests and _record_name_filter parameter
 func (mrb *MaxResultsRequestBuilder) BuildRequest(t ibclient.RequestType, obj ibclient.IBObject, ref string, queryParams ibclient.QueryParams) (req *http.Request, err error) {
 	req, err = mrb.WapiRequestBuilder.BuildRequest(t, obj, ref, queryParams)
 	if req.Method == "GET" {
-		query := req.URL.Query()
-		query.Set("_max_results", strconv.Itoa(mrb.maxResults))
-		req.URL.RawQuery = query.Encode()
+		if mrb.maxResults > 0 {
```

Comment:
My suggestion would be to join these to if statements.
Less depth and personal preference.
```
if req.Method == "GET" && mrb.maxResults > 0 {
}
```

Reply:
Commit-ed suggestion 2.